FWCore/Framework/test/BuildFile.xml

```diff
@@ -36,6 +36,11 @@
   <use name="FWCore/Utilities"/>
 </bin>
 
+<bin name="TestFWCoreFrameworkOptions" file="TestDriver.cpp">
```

Comment:
There is scram support now for specifying running a `.sh` script as part of a test without doing the `TestDriver.cpp` approach. I suggest switching to that since it reduces the number of executables we need.

Reply:
@smuzaffar I am not familiar with this (I've never seen it or forgot...). Is there is an example or documentation somewhere? Does it work the same way and set the same environmental variables as the old way? Here is the script I want to run:

```
#!/bin/bash

 function die { echo Failure $1: status $2 ; exit $2 ; }

 pushd ${LOCAL_TMP_DIR}

 echo testOptions1_cfg.py
 cmsRun -p ${LOCAL_TEST_DIR}/testOptions1_cfg.py >& testOptions1.log || die "cmsRun testOptions1_cfg.py" $?
 grep "Number of Streams = 1" testOptions1.log || die "Failed number of streams test" $?
 grep "Number of Concurrent Lumis = 1" testOptions1.log || die "Failed number of concurrent lumis test" $?
 grep "Number of Concurrent IOVs = 1" testOptions1.log || die "Failed number of concurrent IOVs test" $?

 echo testOptions2_cfg.py
 cmsRun -p ${LOCAL_TEST_DIR}/testOptions2_cfg.py >& testOptions2.log || die "cmsRun testOptions2_cfg.py" $?
 grep "Number of Streams = 5" testOptions2.log || die "Failed number of streams test" $?
 grep "Number of Concurrent Lumis = 4" testOptions2.log || die "Failed number of concurrent lumis test" $?
 grep "Number of Concurrent IOVs = 3" testOptions2.log || die "Failed number of concurrent IOVs test" $?

 echo testOptions3_cfg.py
 cmsRun -p ${LOCAL_TEST_DIR}/testOptions3_cfg.py >& testOptions3.log || die "cmsRun testOptions3_cfg.py" $?
 grep "Number of Streams = 6" testOptions3.log || die "Failed number of streams test" $?
 grep "Number of Concurrent Lumis = 2" testOptions3.log || die "Failed number of concurrent lumis test" $?
 grep "Number of Concurrent IOVs = 2" testOptions3.log || die "Failed number of concurrent IOVs test" $?

 echo testOptions4_cfg.py
 cmsRun -p ${LOCAL_TEST_DIR}/testOptions4_cfg.py >& testOptions4.log || die "cmsRun testOptions4_cfg.py" $?
 grep "Number of Streams = 6" testOptions4.log || die "Failed number of streams test" $?
 grep "Number of Concurrent Lumis = 6" testOptions4.log || die "Failed number of concurrent lumis test" $?
 grep "Number of Concurrent IOVs = 6" testOptions4.log || die "Failed number of concurrent IOVs test" $?

 echo testOptions5_cfg.py
 cmsRun -p ${LOCAL_TEST_DIR}/testOptions5_cfg.py >& testOptions5.log || die "cmsRun testOptions5_cfg.py" $?
 grep "Number of Streams = 1" testOptions5.log || die "Failed number of streams test" $?
 grep "Number of Concurrent Lumis = 1" testOptions5.log || die "Failed number of concurrent lumis test" $?
 grep "Number of Concurrent IOVs = 1" testOptions5.log || die "Failed number of concurrent IOVs test" $?

 echo testOptions6_cfg.py
 cmsRun -p ${LOCAL_TEST_DIR}/testOptions6_cfg.py >& testOptions6.log || die "cmsRun testOptions6_cfg.py" $?
 # Cannot run the grep tests because by default the options are not dumped.
 # You can however run this manually with a debugger and check (which was done)
 # And also just run it and see it doesn't crash...

 rm testOptions1.log
 rm testOptions2.log
 rm testOptions3.log
 rm testOptions4.log
 rm testOptions5.log
 rm testOptions6.log

 popd

 exit 0
```

Reply:
Here
https://github.com/cms-sw/cmssw/blob/98e05a986b61d661b6ddc653f1978bc12fed4278/FWCore/SharedMemory/test/BuildFile.xml#L17-L19

https://github.com/cms-sw/cmssw/blob/98e05a986b61d661b6ddc653f1978bc12fed4278/FWCore/Framework/test/BuildFile.xml#L393-L395

https://github.com/cms-sw/cmssw/blob/98e05a986b61d661b6ddc653f1978bc12fed4278/FWCore/Integration/test/BuildFile.xml#L447-L451

Reply:
@wddgit  , it works nearly the same way ( see examples of it https://github.com/cms-sw/cmssw/blob/master/PhysicsTools/PythonAnalysis/test/BuildFile.xml ) but it does not set the env variables set here https://github.com/cms-sw/cmssw/blob/master/FWCore/Utilities/src/TestHelper.cc#L149-L164 . But of of these you can calculate within the script itself. So basically you just need to use
```
<test name="TestFWCoreFrameworkOptions" command="your-script-in-test.sh args"/>
```

Reply:
@wddgit , if the tests in your script do not depend on each other then better to have a small script e.g.
- `runtest.sh`
```
 echo $1
 cmsRun -p ${CMSSW_BASE}/src/FWCore/Framework/test/$1 >& ${1}.log || die "cmsRun$1" $?
 grep "Number of Streams = $2" ${1}.log || die "Failed number of streams test" $?
 grep "Number of Concurrent Lumis = $3" ${1}.log || die "Failed number of concurrent lumis test" $?
 grep "Number of Concurrent IOVs = $4" ${1}.log || die "Failed number of concurrent IOVs test" $?
```

and then add multiple tests e.g.
```
<test name="TestFWCoreFrameworkOptions1" command="runtest.sh testOptions1_cfg.py 1 1 1"/>
<test name="TestFWCoreFrameworkOptions2" command="runtest.sh testOptions2_cfg.py 5 4 3"/>
...
```

this way scram can run these in parallel. Only thing you need to make sure that these tests do not write in to same output file

Reply:
I mostly did what was suggested here. The main difference is instead of putting so much in the BuildFile I passed only a single index to identify the test and put the config file names and expected results in arrays in the bash script. It seemed to be too much to put expected results in a BuildFile.

Overall, this seems much better. Thanks.